client/src/redux/actions/signature.js

```diff
@@ -7,10 +7,24 @@ export function addsignature(data) {
   };
 }
 
-export default function fetchSignatures(campaignId) {
+export function removeSignature(signatureId) {
   return {
-    type: 'FETCH_SIGNATURES',
-    promise: createApiRequest(`api/signatures/${campaignId}`, 'GET')
+    type: 'REMOVE_SIGNATURE_FROM_FROM_CAMPAIGN',
+    promise: createApiRequest(`api/signatures/${signatureId}`, 'DELETE')
+  };
+}
+
+export function fetchCampaignSignatures(campaignId) {
+  return {
+    type: 'FETCH_CAMPAIGN_SIGNATURES',
+    promise: createApiRequest(`api/signatures/campaigns/${campaignId}`, 'GET')
+  };
+}
+
+export function fetchUserSignatures(userId) {
+  return {
+    type: 'FETCH_USER_SIGNATURES',
+    promise: createApiRequest(`api/signatures/users/${userId}`, 'GET')
```

Comment:
Getting duplicates for FETCH_USER_SIGNATURES and FETCH_USER_SIGNATURES_SUCCESS

<img width="1276" alt="screen shot 2018-04-12 at 15 51 38" src="https://user-images.githubusercontent.com/25015764/38706165-7e9c547c-3e69-11e8-9002-f11e06e504f5.png">

Reply:
I think this is related to the fact that both NavBar and SignCampaign rely on userSignature and when the page is opened on a campaign both fetchUserSignatures() are being called, one by each.  Not 100% sure though

Reply:
Are you solving this or want to make an issue to clean this up? Don't think it's blocking us at the moment but would like to make it more succinct.